spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/stats/LoadBalancerTags.java

```diff
@@ -25,17 +25,21 @@
 import org.springframework.cloud.client.loadbalancer.RequestDataContext;
 import org.springframework.cloud.client.loadbalancer.ResponseData;
 import org.springframework.util.StringUtils;
+import org.springframework.web.reactive.function.client.WebClient;
 
 /**
  * Utility class for building metrics tags for load-balanced calls.
  *
  * @author Olga Maciaszek-Sharma
+ * @author Jaroslaw Dembek
  * @since 3.0.0
  */
 final class LoadBalancerTags {
 
 	static final String UNKNOWN = "UNKNOWN";
 
+	static final String URI_TEMPLATE_ATTRIBUTE = WebClient.class.getName() + ".uriTemplate";
```

Comment:
Load-balanced requests can come from `WebClient`, but also `RestClient` and `RestTemplate`, so we need a way to handle all those scenarios.

Reply:
`RestClient` is new to me and I almost forget about `RestTemplate` existence. 
`WebClient` has two impls: MVC and WebFlux.
My change is for sure working in SpringBoot 3.1 + WebFlux.
I would start with writing test for all 4 scenarios. Could you advise me on pattern (code example of integration test setup e.g. some old or new PR) I should follow?

Reply:
https://github.com/spring-projects/spring-framework/issues/31882
There is a change in Spring about value of URI_TEMPLATE_ATTRIBUTE - now it will include `baseUrl`. Which for load balanced calls is `http://{service-name}/...`

Reply:
Since for a given client the `baseUrl` part of the template  is going to remain the same anyhow, it should not be an issue when it comes to measuring the requests - the metrics will be effectively differentiated by the path. @marcingrzejszczak wdyt?

Reply:
I'm not sure what you mean by "`WebClient` has two implementations"? Spring supports blocking or MVC style communication, which is implemented by `RestTemplate` and `RestClient`, and reactive communication, implemented by `WebClient`.

We have integration tests that verify that `LoadBalancerLifecycle` calls are correctly executed. I'm thinking you could use a similar pattern to work on your integration test. Here are the test classes (they cover blocking load-balancing with and without retries and reactive load-balancing with and without retries):
- `BlockingLoadBalancerClientTests`
- `RetryLoadBalancerInterceptorTests`
- `ReactorLoadBalancerExchangeFilterFunctionTests`
- `RetryableLoadBalancerExchangeFilterFunctionIntegrationTests`
Let me know if you need more help with this. If you do not have time and would like us to take over the PR to work on the non-reactive part, also no worries - let me know.